xstate/scxml.py

```diff
@@ -1,33 +1,82 @@
-from typing import List
-from xstate.transition import Transition
-
-# defaultHistoryContent = new HashTable()
-# computeEntrySet(enabledTransitions, statesToEnter, statesForDefaultEntry, defaultHistoryContent)
-# for s in statesToEnter.toList().sort(entryOrder):
-#     configuration.add(s)
-#     statesToInvoke.add(s)
-#     if binding == "late" and s.isFirstEntry:
-#         initializeDataModel(datamodel.s,doc.s)
-#         s.isFirstEntry = false
-#     for content in s.onentry.sort(documentOrder):
-#         executeContent(content)
-#     if statesForDefaultEntry.isMember(s):
-#         executeContent(s.initial.transition)
-#     if defaultHistoryContent[s.id]:
-#         executeContent(defaultHistoryContent[s.id])
-#     if isFinalState(s):
-#         if isSCXMLElement(s.parent):
-#             running = false
-#         else:
-#             parent = s.parent
-#             grandparent = parent.parent
-#             internalQueue.enqueue(new Event("done.state." + parent.id, s.donedata))
-#             if isParallelState(grandparent):
-#                 if getChildStates(grandparent).every(isInFinalState):
-#                     internalQueue.enqueue(new Event("done.state." + grandparent.id))
-
-
-def enter_states(enabled_transitions: List[Transition]):
-    states_to_enter = set()
-    states_for_default_entry = set()
+import xml.etree.ElementTree as ET
+from typing import Optional
+import json
 
+from xstate.machine import Machine
+
+ns = {"scxml": "http://www.w3.org/2005/07/scxml"}
```

Comment:
This was necessary because even though in XML it shows as e.g., `<state>`, when it's parsed, the name shows up as `(http://www.w3.org/2005/07/scxml)state`, which is really annoying. So we have to tell the parser what namespaces we're using.

Had to spend an hour figuring this out 😅